test/unit/puppetlabs/services/jruby/jruby_puppet_core_test.clj

```diff
@@ -13,6 +14,37 @@
     :ruby-load-path ["./ruby/puppet/lib" "./ruby/facter/lib"]},
    :certificate-authority {:certificate-status {:client-whitelist []}}})
 
+(defmacro with-stdin-str
+  "Evaluates body in a context in which System/in is bound to a fresh
+  input stream initialized with the string s.  The return value of evaluating
+  body is returned."
+  [s & body]
+  `(let [system-input# (System/in)
+         string-input# (new ByteArrayInputStream (.getBytes ~s))]
+     (try
+       (System/setIn string-input#)
+       ~@body
+       (finally (System/setIn system-input#)))))
+
+(defmacro capture-out
+  "capture System.out and return it as the value of :out in the return map.
+  The return value of body is available as :return in the return map.
+
+  This macro is not thread safe."
+  [& body]
+  `(let [return-map# (atom {})
+         system-output# (System/out)
+         captured-output# (new ByteArrayOutputStream)
+         capturing-print-stream# (new PrintStream captured-output#)]
+     (try
+       (System/setOut capturing-print-stream#)
```

Comment:
I think that in Clojure you might normally do this with something like:

`(binding [*out* capturing-print-stream#]`

But, not sure if that would work for JRuby, so if we're messing w/stdin and stdout at all during the tests, it probably doesn't matter much which way we do it.

I wonder if it would be worth putting these tests into the `integration` test suite since they're a little riskier (because they much with stdout/stderr)?  Not sure about that one either.